src/any_agent/evaluation/runner.py

```diff
@@ -0,0 +1,145 @@
+import json
+from textwrap import dedent
+from typing import Any
+
+from any_agent.evaluation.evaluators import (
+    CheckpointEvaluator,
+    HypothesisEvaluator,
+    LLMEvaluator,
+    QuestionAnsweringSquadEvaluator,
+)
+from any_agent.evaluation.evaluators.schemas import EvaluationResult
+from any_agent.evaluation.results_saver import save_evaluation_results
+from any_agent.evaluation.test_case import TestCase
+from any_agent.logging import logger
+from any_agent.telemetry import TelemetryProcessor
+
+
+class EvaluationRunner:
+    def __init__(self):
+        self._test_cases: list[TestCase] = []
+        self._telemetry_paths: list[str] = []
+        self.checkpoint_evaluator: CheckpointEvaluator | None = None
+        self.hypothesis_evaluator: HypothesisEvaluator | None = None
+        self.qa_evaluator: QuestionAnsweringSquadEvaluator | None = None
+
+    def _setup_evaluators(self, test_case: TestCase) -> None:
+        self.checkpoint_evaluator = CheckpointEvaluator(model=test_case.llm_judge)
+        self.hypothesis_evaluator = HypothesisEvaluator(model=test_case.llm_judge)
+        self.qa_evaluator = QuestionAnsweringSquadEvaluator()
+
+    def add_test_case(self, test_case_path: str) -> None: ...
+
+    def add_telemetry(self, telemetry_path: str) -> None: ...
+
+    def _run_telemetry_eval(self, test_case: TestCase, telemetry_path: str) -> None:
+        with open(telemetry_path) as f:
+            telemetry: list[dict[str, Any]] = json.loads(f.read())
+        logger.info(f"Telemetry loaded from {telemetry_path}")
+
+        agent_framework = TelemetryProcessor.determine_agent_framework(telemetry)
+
+        processor = TelemetryProcessor.create(agent_framework)
+        hypothesis_answer = processor._extract_hypothesis_answer(trace=telemetry)
```

Comment:
IMO, we should decouple the processor logic that we use to generate console outputs from any evaluation logic. 

Maybe we need to prioritize what I commented in https://github.com/mozilla-ai/any-agent/pull/183#pullrequestreview-2807512775 and have a standard `AgentTrace` class that can be loaded from the trace file.

Then, evaluation works from there and the `extract_stuff` for evaluation should be implemented in the `evaluation` module

Reply:
This is a good comment. I will keep this in mind while I work on #183 💯

Reply:
Ok cool I think I have it implemented in #183, take a look at that and let me know what you think. Sorry it's such a big PR now 🙈